Sync/State.swift

```diff
@@ -473,6 +463,11 @@ open class Scratchpad {
 
         b.hashedUID = prefs.stringForKey(PrefHashedUID)
 
+        b.fxaDeviceId = prefs.stringForKey(PrefFxADeviceId) ?? {
+            log.warning("No value found in prefs for fxaDeviceId! Will overwrite on first sync")
+            return "unknown_fxaDeviceId"
```

Comment:
Almost all clients will have the old method — a JSON-encoded bundle representing the entire registration. That's the v1 format.

You should be looking for that here.

I'm in two minds whether to ask you to define a V2 format that stores the device ID, instead of hacking this in here.

Reply:
Added a migration and a comment here.